cdap-app-templates/cdap-etl/cdap-data-pipeline/src/main/java/co/cask/cdap/datapipeline/SmartWorkflow.java

```diff
@@ -581,7 +581,9 @@ public void destroy() {
       Map<String, List<FieldOperation>> stageOperations
         = GSON.fromJson(nodeValue.getValue().toString(), STAGE_OPERATIONS_MAP);
       for (Map.Entry<String, List<FieldOperation>> entry : stageOperations.entrySet()) {
-        allStageOperations.get(entry.getKey()).addAll(entry.getValue());
+        if (allStageOperations.containsKey(entry.getKey())) {
```

Comment:
`allStageOperations` - is the name misleading, if it doesn't actually contain all the stage operations?

Reply:
It contains all the stages which is visible to the users from the StageSpecs, but does not contain the connector stage, I will add a comment about it.